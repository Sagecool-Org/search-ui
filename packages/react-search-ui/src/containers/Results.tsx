import React from "react";
import {
  Result,
  Results,
  ResultsContainerProps
} from "@elastic/react-search-ui-views";

import { Result as ResultContainer } from ".";
import { useSearch } from "../hooks";

function getRaw(result, value) {
  if (!result[value] || !result[value].raw) return;
  return result[value].raw;
}

const ResultsContainer = ({
  clickThroughTags = [],
  resultView,
  shouldTrackClickThrough = true,
  titleField,
  urlField,
  thumbnailField,
  view,
  ...rest
}: ResultsContainerProps) => {
  const { results } = useSearch();

  const View = view || Results;
  const ResultView = resultView || Result;

  const children = results.map((result) => (
    <ResultContainer
      key={`result-${getRaw(result, "id")}`}
      titleField={titleField}
      urlField={urlField}
      thumbnailField={thumbnailField}
      view={ResultView}
      shouldTrackClickThrough={shouldTrackClickThrough}
      clickThroughTags={clickThroughTags}
      result={result}
    />
  ));
  const viewProps = {
    children,
    ...rest
  };

  return <View {...viewProps} />;
};

export default ResultsContainer;
