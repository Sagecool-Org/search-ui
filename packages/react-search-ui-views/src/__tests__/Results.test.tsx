import React from "react";
import { render } from "@testing-library/react";
import Results from "../Results";

it("renders correctly", () => {
  const { container } = render(
    <Results>
      <div>Children</div>
    </Results>
  );
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <Results className={customClassName}>
      <div>Children</div>
    </Results>
  );
  expect(container.firstChild).toHaveClass(
    "sui-results-container",
    "test-class"
  );
});
