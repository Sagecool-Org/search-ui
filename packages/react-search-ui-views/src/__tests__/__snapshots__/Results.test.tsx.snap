// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    class="sui-results-container"
  >
    <div>
      Children
    </div>
  </ul>
</div>
`;
